[32mINFO    [0m integration.ha_tests.test_restore_cluster:test_restore_cluster.py:80 creating a table in the database
[32mINFO    [0m integration.ha_tests.test_restore_cluster:test_restore_cluster.py:88 Downscaling the existing cluster
[32mINFO    [0m integration.ha_tests.test_restore_cluster:test_restore_cluster.py:97 Upscaling the second cluster with the old data
[32mINFO    [0m juju.model:model.py:2971 Waiting for model:
  second-cluster/1 [allocating] waiting: waiting for machine
[32mINFO    [0m juju.model:model.py:2971 Waiting for model:
  second-cluster/1 [allocating] waiting: waiting for machine
[32mINFO    [0m juju.model:model.py:2971 Waiting for model:
  second-cluster/1 [allocating] waiting: agent initialising
[32mINFO    [0m juju.model:model.py:2971 Waiting for model:
  second-cluster/1 [executing] waiting: waiting to start PostgreSQL
[32mINFO    [0m juju.model:model.py:2971 Waiting for model:
  second-cluster/1 [executing] waiting: waiting to start PostgreSQL
[32mINFO    [0m juju.model:model.py:2971 Waiting for model:
  second-cluster/1 [executing] waiting: waiting to start PostgreSQL
[32mINFO    [0m juju.model:model.py:2971 Waiting for model:
  second-cluster/1 [executing] waiting: waiting to start PostgreSQL
[32mINFO    [0m juju.model:model.py:2971 Waiting for model:
  second-cluster/1 [executing] waiting: waiting to start PostgreSQL
[32mINFO    [0m juju.model:model.py:2971 Waiting for model:
  second-cluster/1 [executing] waiting: waiting to start PostgreSQL
[32mINFO    [0m juju.model:model.py:2971 Waiting for model:
  second-cluster/1 [executing] waiting: waiting to start PostgreSQL
[32mINFO    [0m juju.model:model.py:2971 Waiting for model:
  second-cluster/1 [executing] waiting: waiting for primary to be reachable from this unit
[32mINFO    [0m juju.model:model.py:2971 Waiting for model:
  second-cluster/1 [executing] waiting: waiting for primary to be reachable from this unit
[32mINFO    [0m juju.model:model.py:2971 Waiting for model:
  second-cluster/1 [executing] waiting: waiting for primary to be reachable from this unit
[32mINFO    [0m juju.model:model.py:2971 Waiting for model:
  second-cluster/1 [executing] active: Primary
[32mINFO    [0m juju.model:model.py:2971 Waiting for model:
  second-cluster/1 [idle] active: Primary
[32mINFO    [0m juju.model:model.py:2971 Waiting for model:
  second-cluster/1 [idle] active: Primary
  second-cluster/2 [allocating] waiting: waiting for machine
[32mINFO    [0m juju.model:model.py:2971 Waiting for model:
  second-cluster/2 [allocating] waiting: waiting for machine
[32mINFO    [0m juju.model:model.py:2971 Waiting for model:
  second-cluster/2 [allocating] waiting: agent initialising
[32mINFO    [0m juju.model:model.py:2971 Waiting for model:
  second-cluster/1 [executing] active: Primary
  second-cluster/2 [executing] waiting: waiting to start PostgreSQL
[32mINFO    [0m juju.model:model.py:2971 Waiting for model:
  second-cluster/1 [executing] maintenance: reconfiguring cluster
  second-cluster/2 [executing] waiting: waiting to start PostgreSQL
[32mINFO    [0m juju.model:model.py:2971 Waiting for model:
  second-cluster/1 [idle] active: Primary
  second-cluster/2 [executing] waiting: waiting to start PostgreSQL
[32mINFO    [0m juju.model:model.py:2971 Waiting for model:
  second-cluster/1 [executing] maintenance: reconfiguring cluster
  second-cluster/2 [executing] waiting: awaiting for member to start
[32mINFO    [0m juju.model:model.py:2971 Waiting for model:
  second-cluster/2 [executing] waiting: waiting for primary to be reachable from this unit
[32mINFO    [0m juju.model:model.py:2971 Waiting for model:
  second-cluster/1 [executing] active: Primary
  second-cluster/2 [idle] active: 
[32mINFO    [0m juju.model:model.py:2971 Waiting for model:
  second-cluster/1 [idle] active: Primary
  second-cluster/2 [idle] active: 
  second-cluster/3 [allocating] waiting: waiting for machine
[32mINFO    [0m juju.model:model.py:2971 Waiting for model:
  second-cluster/3 [allocating] waiting: waiting for machine
[32mINFO    [0m juju.model:model.py:2971 Waiting for model:
  second-cluster/3 [allocating] waiting: waiting for machine
[32mINFO    [0m juju.model:model.py:2971 Waiting for model:
  second-cluster/3 [executing] maintenance: installing PostgreSQL
[32mINFO    [0m juju.model:model.py:2971 Waiting for model:
  second-cluster/3 [executing] maintenance: installing PostgreSQL
[32mINFO    [0m juju.model:model.py:2971 Waiting for model:
  second-cluster/3 [executing] maintenance: installing PostgreSQL
[32mINFO    [0m juju.model:model.py:2971 Waiting for model:
  second-cluster/3 [executing] maintenance: installing PostgreSQL
[32mINFO    [0m juju.model:model.py:2971 Waiting for model:
  second-cluster/3 [executing] maintenance: installing PostgreSQL
[32mINFO    [0m pytest_operator.plugin:plugin.py:862 Model status:

Model  Controller           Cloud/Region         Version  SLA          Timestamp
test   localhost-localhost  localhost/localhost  3.4.4    unsupported  02:16:35Z

App             Version  Status  Scale  Charm       Channel  Rev  Exposed  Message
second-cluster           active      3  postgresql             1  no       

Unit               Workload     Agent      Machine  Public address  Ports     Message
second-cluster/1*  maintenance  executing  4        10.4.107.113    5432/tcp  reconfiguring cluster
second-cluster/2   active       executing  5        10.4.107.61     5432/tcp  
second-cluster/3   error        idle       6        10.4.107.58               hook failed: "config-changed"

Machine  State    Address       Inst id        Base          AZ  Message
4        started  10.4.107.113  juju-4801c6-4  ubuntu@22.04      Running
5        started  10.4.107.61   juju-4801c6-5  ubuntu@22.04      Running
6        started  10.4.107.58   juju-4801c6-6  ubuntu@22.04      Running

[32mINFO    [0m pytest_operator.plugin:plugin.py:868 Juju error logs:

machine-2: 01:51:24 ERROR juju.worker.dependency "lxd-container-provisioner" manifold worker returned unexpected error: container types not yet available
machine-2: 01:51:24 ERROR juju.worker.dependency "kvm-container-provisioner" manifold worker returned unexpected error: container types not yet available
unit-first-cluster-2: 01:51:25 ERROR juju.worker.meterstatus error running "meter-status-changed": charm missing from disk
machine-0: 01:51:32 ERROR juju.worker.dependency "kvm-container-provisioner" manifold worker returned unexpected error: container types not yet available
machine-0: 01:51:32 ERROR juju.worker.dependency "lxd-container-provisioner" manifold worker returned unexpected error: container types not yet available
unit-first-cluster-0: 01:51:32 ERROR juju.worker.meterstatus error running "meter-status-changed": charm missing from disk
machine-1: 01:51:32 ERROR juju.worker.dependency "lxd-container-provisioner" manifold worker returned unexpected error: container types not yet available
machine-1: 01:51:33 ERROR juju.worker.dependency "kvm-container-provisioner" manifold worker returned unexpected error: container types not yet available
unit-first-cluster-1: 01:51:33 ERROR juju.worker.meterstatus error running "meter-status-changed": charm missing from disk
machine-3: 01:51:49 ERROR juju.worker.dependency "kvm-container-provisioner" manifold worker returned unexpected error: container types not yet available
machine-3: 01:51:49 ERROR juju.worker.dependency "lxd-container-provisioner" manifold worker returned unexpected error: container types not yet available
unit-second-cluster-0: 01:51:50 ERROR juju.worker.meterstatus error running "meter-status-changed": charm missing from disk
machine-4: 02:02:05 ERROR juju.worker.dependency "kvm-container-provisioner" manifold worker returned unexpected error: container types not yet available
machine-4: 02:02:05 ERROR juju.worker.dependency "lxd-container-provisioner" manifold worker returned unexpected error: container types not yet available
unit-second-cluster-1: 02:02:05 ERROR juju.worker.meterstatus error running "meter-status-changed": charm missing from disk
machine-5: 02:09:25 ERROR juju.worker.dependency "kvm-container-provisioner" manifold worker returned unexpected error: container types not yet available
machine-5: 02:09:25 ERROR juju.worker.dependency "lxd-container-provisioner" manifold worker returned unexpected error: container types not yet available
unit-second-cluster-2: 02:09:25 ERROR juju.worker.meterstatus error running "meter-status-changed": charm missing from disk
machine-6: 02:13:52 ERROR juju.worker.dependency "lxd-container-provisioner" manifold worker returned unexpected error: container types not yet available
machine-6: 02:13:52 ERROR juju.worker.dependency "kvm-container-provisioner" manifold worker returned unexpected error: container types not yet available
unit-second-cluster-3: 02:13:53 ERROR juju.worker.meterstatus error running "meter-status-changed": charm missing from disk
unit-second-cluster-3: 02:16:29 ERROR unit.second-cluster/3.juju-log An exception occurred when installing charmed-postgresql. Reason: Snap: 'charmed-postgresql'; command ['snap', 'install', 'charmed-postgresql', '--channel="14/stable"', '--revision="120"'] failed with output = ''
unit-second-cluster-3: 02:16:35 ERROR unit.second-cluster/3.juju-log Uncaught exception while in charm code:
Traceback (most recent call last):
  File "/var/lib/juju/agents/unit-second-cluster-3/charm/./src/charm.py", line 1849, in <module>
    main(PostgresqlOperatorCharm)
  File "/var/lib/juju/agents/unit-second-cluster-3/charm/venv/ops/main.py", line 548, in main
    manager.run()
  File "/var/lib/juju/agents/unit-second-cluster-3/charm/venv/ops/main.py", line 527, in run
    self._emit()
  File "/var/lib/juju/agents/unit-second-cluster-3/charm/venv/ops/main.py", line 516, in _emit
    _emit_charm_event(self.charm, self.dispatcher.event_name)
  File "/var/lib/juju/agents/unit-second-cluster-3/charm/venv/ops/main.py", line 147, in _emit_charm_event
    event_to_emit.emit(*args, **kwargs)
  File "/var/lib/juju/agents/unit-second-cluster-3/charm/venv/ops/framework.py", line 348, in emit
    framework._emit(event)
  File "/var/lib/juju/agents/unit-second-cluster-3/charm/venv/ops/framework.py", line 860, in _emit
    self._reemit(event_path)
  File "/var/lib/juju/agents/unit-second-cluster-3/charm/venv/ops/framework.py", line 950, in _reemit
    custom_handler(event)
  File "/var/lib/juju/agents/unit-second-cluster-3/charm/lib/charms/tempo_k8s/v1/charm_tracing.py", line 647, in wrapped_function
    return callable(*args, **kwargs)  # type: ignore
  File "/var/lib/juju/agents/unit-second-cluster-3/charm/./src/charm.py", line 943, in _on_config_changed
    self._validate_config_options()
  File "/var/lib/juju/agents/unit-second-cluster-3/charm/lib/charms/tempo_k8s/v1/charm_tracing.py", line 647, in wrapped_function
    return callable(*args, **kwargs)  # type: ignore
  File "/var/lib/juju/agents/unit-second-cluster-3/charm/./src/charm.py", line 1682, in _validate_config_options
    self.config.instance_default_text_search_config
  File "/var/lib/juju/agents/unit-second-cluster-3/charm/lib/charms/data_platform_libs/v0/data_models.py", line 200, in config
    return self.config_type(**translated_keys)
  File "/var/lib/juju/agents/unit-second-cluster-3/charm/venv/pydantic/main.py", line 339, in __init__
    values, fields_set, validation_error = validate_model(__pydantic_self__.__class__, data)
  File "/var/lib/juju/agents/unit-second-cluster-3/charm/venv/pydantic/main.py", line 1074, in validate_model
    v_, errors_ = field.validate(value, values, loc=field.alias, cls=cls_)
  File "/var/lib/juju/agents/unit-second-cluster-3/charm/venv/pydantic/fields.py", line 895, in validate
    v, errors = self._apply_validators(v, values, loc, cls, self.post_validators)
  File "/var/lib/juju/agents/unit-second-cluster-3/charm/venv/pydantic/fields.py", line 1154, in _apply_validators
    v = validator(cls, v, values, self, self.model_config)
  File "/var/lib/juju/agents/unit-second-cluster-3/charm/venv/pydantic/class_validators.py", line 304, in <lambda>
    return lambda cls, v, values, field, config: validator(cls, v)
  File "/var/lib/juju/agents/unit-second-cluster-3/charm/src/config.py", line 246, in response_lc_values
    output = subprocess.check_output(["ls", "/snap/charmed-postgresql/current/usr/lib/locale"])
  File "/usr/lib/python3.10/subprocess.py", line 421, in check_output
    return run(*popenargs, stdout=PIPE, timeout=timeout, check=True,
  File "/usr/lib/python3.10/subprocess.py", line 526, in run
    raise CalledProcessError(retcode, process.args,
subprocess.CalledProcessError: Command '['ls', '/snap/charmed-postgresql/current/usr/lib/locale']' returned non-zero exit status 2.
unit-second-cluster-3: 02:16:35 ERROR juju.worker.uniter.operation hook "config-changed" (via hook dispatching script: dispatch) failed: exit status 1

[32mINFO    [0m pytest_operator.plugin:plugin.py:947 Forgetting model main...